iati/tests/test_validator.py

```diff
@@ -880,6 +867,23 @@ def test_basic_validation_ruleset_invalid(self, schema_ruleset, invalid_xml_file
         assert iati.validator.is_iati_xml(data, schema_ruleset)
         assert not iati.validator.is_valid(data, schema_ruleset)
 
+    def test_one_ruleset_error_added_for_multiple_rule_errors(self, schema_ruleset):
+        """Check that a dataset containing multiple rule errors produces an error log containing only one ruleset error."""
+        data_with_multiple_rule_errors = iati.core.tests.utilities.load_as_dataset('invalid_std_ruleset_multiple_rule_errors')
+        result = iati.validator.full_validation(data_with_multiple_rule_errors, schema_ruleset)
+
+        assert len(result.get_errors_or_warnings_by_category('rule')) > 1
+        assert len(result.get_errors_or_warnings_by_name('err-ruleset-conformance-fail')) == 1
```

Comment:
What happens if a Schema has multiple Rulesets attached?

Reply:
If a Schema has multiple Rulesets attached, then one `err-ruleset-conformance-fail` would be added for every Ruleset failure. In this case, `schema_ruleset` only contains one ruleset, so I'd say that the 'assert there is only one ruleset error' is correct.

Reply:
Test added in 8051b0a0c9fb1ed3ce2ac66f3125434588e1f746